pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -276,9 +276,6 @@ func (rq *ResourceQuotaController) syncResourceQuota(resourceQuota api.ResourceQ
 	dirty = dirty || (resourceQuota.Status.Hard == nil || resourceQuota.Status.Used == nil)
 
 	used := api.ResourceList{}
-	if resourceQuota.Status.Used != nil {
-		used = quota.Add(api.ResourceList{}, resourceQuota.Status.Used)
-	}
 	hardLimits := quota.Add(api.ResourceList{}, resourceQuota.Spec.Hard)
 
 	newUsage, err := quota.CalculateUsage(resourceQuota.Namespace, resourceQuota.Spec.Scopes, hardLimits, rq.registry)
```

Comment:
to fix the test case that you demonstrate, ideally we want the set of used values to match the set of hard values.

a better fix is to do this after line 288:

```
        // ensure set of used values match those that have hard constraints
	hardResources := quota.ResourceNames(hardLimits)
	used = quota.Mask(used, hardResources)
```